arangod/Pregel/PregelFeature.cpp

```diff
@@ -302,17 +322,10 @@ void PregelFeature::cleanupConductor(uint64_t executionNumber) {
 }
 
 void PregelFeature::cleanupWorker(uint64_t executionNumber) {
-  // make sure no one removes the PregelFeature while in use
-  std::shared_ptr<PregelFeature> instance = ::instance;
-
-  if (!instance) {
-    THROW_ARANGO_EXCEPTION(TRI_ERROR_SHUTTING_DOWN);
-  }
-
   // unmapping etc might need a few seconds
   TRI_ASSERT(SchedulerFeature::SCHEDULER != nullptr);
   Scheduler* scheduler = SchedulerFeature::SCHEDULER;
-  bool queued = scheduler->queue(RequestLane::INTERNAL_LOW, [this, executionNumber, instance] {
+  bool queued = scheduler->queue(RequestLane::INTERNAL_LOW, [this, executionNumber] {
```

Comment:
`SchedulerFeature::stop` sets a flag so no new jobs are accepted and blocks until all pending jobs are processed. That's why we cancel all conductors and workers in `beginShutdown`, so that any cleanup jobs can still be scheduled. `PregelFeature::unprepare` is called after `SchedulerFeature::stop`, so all tasks are already processed. `PregelFeature::stop` is a noop.